clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -817,10 +817,14 @@ protected void close(Timer timer) {
     }
 
     /**
-     * Leave the current group and reset local generation/memberId.
+     * Leave the current group and reset local generation/member id.
      */
     public synchronized void maybeLeaveGroup() {
-        if (!coordinatorUnknown() && state != MemberState.UNJOINED && generation.hasMemberId()) {
+        // Starting from 2.3, only dynamic members will send LeaveGroupRequest to the broker,
```

Comment:
I quickly checked `unsubscribe()` use cases, and there are mainly two:
1. illegal topic/partition data, i.e empty topic partitions to subscribe
2. consumer self managed membership (subscription)

I think it makes sense to make static member behavior consistent in these two cases, because the effect of leaving is minimal.

Reply:
I've thought about this a bit more, and also searched in github: https://github.com/search?l=Java&q=%22consumer.unsubscribe%22&type=Code

I think a third common case is to use a temporary consumer for its APIs, like get offset by timestamp, get log end offset etc; generally speaking for temporary consumer case, they should not use static members (and by default it would not be the case). 

So I think it really boils down to: for static members, do we consider the admin request kicking it out of the group be the only appropriate way for it to leave in time or not? I.e. even if the consumer shuts down itself, it should not be considered as "I want to leave" but another request has to be made to effectively kick him out.